packages/core/application/core-application-browser/src/app_mount.ts

```diff
@@ -87,9 +87,6 @@ export interface AppMountParameters<HistoryLocationState = unknown> {
    * The route path for configuring navigation to the application.
    * This string should not include the base path from HTTP.
    *
-   * @deprecated Use {@link AppMountParameters.history} instead.
-   * @removeBy 8.8.0
-   *
```

Comment:
Was this misplaced? Can't we remove `appBasePath`? Is it not really deprecated?

Reply:
I don't think we can remove `appBasePath` for a few reasons:
- `appBasePath` is still used by several plugins e.g [src/plugins/kibana_overview/public/application.tsx:](https://github.com/elastic/kibana/blob/88733fc48f2723569d72c59141788b201e777ddd/src/plugins/kibana_overview/public/application.tsx#L43-L46). 
- the tight dependency between `history` and `react-router` prevents us from updating the `history` package without considerable effort. ATM, a few teams are blocked from using `AppMountParams.history` because of that.
- Since we're not deprecating `onAppLeave` or `AppLeaveHandler`, we need to carry on supporting `appBasePath` too

(@pgayvallet please correct me if I'm wrong

Reply:
IIRC `appBasePath` exposition isn't directly related to the `ScopedHistory.block` thingy. It's exposed because applications based on the legacy (Hash) router needs to know about it to instantiate their own router.  I think nothing would block them from migrating to the new (HTML5) router.

Q/A:
should it be deprecated? ihmo yes.
would it make sense to delete it? ihmo yes.
can we delete it? no, as other teams are still using it...
can we easily adapt the code ourselves to delete it? no, migrating from one router to the other is all but trivial, and multiple apps are still using it IIRC
so, what should we do? not sure. Removing the deprecation may be fine, idk.

Reply:
We have a bit of time to see if teams have the capacity to migrate their routers and don't need to make a decision right now. 
If there's no actual harm in keeping `appBasePath` around then we can treat it in the same way as `onAppLeave`: remove the deprecation warning even though it's not ideal.